armtemplates/template.bicep

```diff
@@ -0,0 +1,280 @@
+  name: virtualMachines_iagora_name
+  location: 'francecentral'
+  zones: [
+    '1'
+  ]
+  properties: {
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    additionalCapabilities: {
+      hibernationEnabled: false
+    }
+    storageProfile: {
+      imageReference: {
+        publisher: 'canonical'
+        offer: '0001-com-ubuntu-server-focal'
+        sku: '20_04-lts-gen2'
+        version: 'latest'
+      }
+      osDisk: {
+        osType: 'Linux'
+        name: '${virtualMachines_iagora_name}_OsDisk_1_bcb7e14b83d24fc7a077197bdf465401'
+        createOption: 'FromImage'
+        caching: 'ReadWrite'
+        managedDisk: {
+          storageAccountType: 'Premium_LRS'
+          /*id: resourceId(
+            'Microsoft.Compute/disks',
+            '${virtualMachines_iagora_name}_OsDisk_1_bcb7e14b83d24fc7a077197bdf465401'
+          )*/
+        }
+        deleteOption: 'Delete'
+        diskSizeGB: 30
+      }
+      dataDisks: []
+      diskControllerType: 'SCSI'
+    }
+    osProfile: {
+      computerName: virtualMachines_iagora_name
+      adminUsername: virtualMachines_iagora_name
+      linuxConfiguration: {
+        disablePasswordAuthentication: true
+        ssh: {
+          publicKeys: [
+            {
+              path: '/home/${virtualMachines_iagora_name}/.ssh/authorized_keys'
+              keyData: sshPublicKey 
+            }
+          ]
+        }
+        provisionVMAgent: true
+        patchSettings: {
+          patchMode: 'AutomaticByPlatform'
+          automaticByPlatformSettings: {
+            rebootSetting: 'IfRequired'
+            bypassPlatformSafetyChecksOnUserSchedule: false
+          }
+          assessmentMode: 'ImageDefault'
+        }
+      }
+      secrets: []
+      allowExtensionOperations: true
+      //requireGuestProvisionSignal: true
+    }
+    securityProfile: {
+      uefiSettings: {
+        secureBootEnabled: true
+        vTpmEnabled: true
+      }
+      securityType: 'TrustedLaunch'
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: networkInterfaces_iagora731_z1_name_resource.id
+          properties: {
+            deleteOption: 'Detach'
+          }
+        }
+      ]
+    }
+    diagnosticsProfile: {
+      bootDiagnostics: {
+        enabled: true
+      }
+    }
+  }
+}
+
+//subnet ressourc
+
+resource virtualNetworks_iagora_vnet_name_default 'Microsoft.Network/virtualNetworks/subnets@2023-11-01' = {
+  name: 'default'
+  parent:virtualNetworks_iagora_vnet_name_resource
+  properties: {
+    addressPrefix: '10.0.0.0/24'
+    delegations: []
+    privateEndpointNetworkPolicies: 'Disabled'
+    privateLinkServiceNetworkPolicies: 'Enabled'
+  }
+  
+}
+
+//interface ressour
+
+resource networkInterfaces_iagora731_z1_name_resource 'Microsoft.Network/networkInterfaces@2023-11-01' = {
+  name: networkInterfaces_iagora731_z1_name
+  location: 'francecentral'
```

Comment:
Shouldn't the location be configurable?

Reply:
Yes, the location should be configurable to provide flexibility in deployment. Allowing the location to be set through parameters or configuration options enables you to deploy resources in different regions based on requirements such as performance, compliance, or cost. Making the location configurable ensures that the deployment can be adapted to various scenarios and needs.

Reply:
Great) I'm happy that we agree) Let's make it configurable then) For example:
```bicep
param location string = resourceGroup().location
```